src/App.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+const App = ({ cells, winner, currentTurn, dispatch }) => {
+  const getCellText = value => value || '';
+  return (
+    <div>
+      <h1 className="Title">
+      Tic Tac Toe
+        { winner ? `(Winner ${winner})` : `(turn ${currentTurn})`}
+      </h1>
+
+      <div className="Ceil-container">
+
+        {cells.map((cell, i) => (
+          <span
+            key={i}
+            className="Cell"
+            onClick={() => dispatch({
```

Comment:
Suggestion: Define `dispatchTurn` function above and call it with `i` as the parameter. 
```
onClick(() => {
  dispatchTurn(i);
})
```